geopandas/base.py

```diff
@@ -3485,6 +3487,211 @@ def equals(self, other):
             return False
         return self._data.equals(other._data)
 
+    def sample_points(
+        self, size=None, spacing=None, method="random", tile=None, **sample_kwargs
+    ):
+        """
+        Sample points from each geometry.
+
+        Generate MultiPoint per each geometry containing points sampled according to the
+        specification. You can either sample randomly, in a square or hexagonal grid or
+        using the advanced sampling algorithm rom the ``pointpats.random`` module.
+
+        For polygons, this samples within the area of the polygon. For lines,
+        this samples along the length of the linestring. For multi-part
+        geometries, the weights of each part are selected according to their relevant
+        attribute (area for Polygons, length for LineStrings), and then points are
+        sampled from each part.
+
+        Any other geometry type (e.g. Point, GeometryCollection) is ignored, and an
+        empty MultiPoint geometry is returned.
+
+        Parameters
+        ----------
+        size : int | tuple | array-like, optional
+
+            The size of the sample requested.
+
+            If ``method='random'``, this indicates the
+            number of samples to draw from each geometry. If a tuple is provided,
+            then the first value must indicate the number of samples for each geometry,
+            and the second value must indicate the number of replications to sample.
```

Comment:
Ah, I see below that this results in different columns?

I am wondering if that is needed? A user can still call this method several times and create those columns themselves? That's a bit more complex for that specific use case, but it would simplify it here (and not have an ambiguous meaning of what a tuple for `size` means)